mitodl/hugo-course-publisher
# add pagination to search

#### Pre-Flight checklist


#### What are the relevant tickets?

part of #178

#### What's this PR do?

this adds pagination to the search, using the same infinite scroll package we're using elsewhere.

#### How should this be manually tested?

go to `/search` and check it out. the infinite scrolling should work as you expect, meaning:

- scrolling down shows you more results for the existing search you already have entered into the search bar
- changing the text in the search bar clears all the existing results and shows you the first page of results for the new search
- there is a spinner shown when the initial search is loading and then also when a new page is loading

#### Screenshots (if appropriate)

basic spinner:

![Screenshot from 2020-08-20 10-56-09](https://user-images.githubusercontent.com/6207644/90788436-cfc51d00-e2d3-11ea-9434-9b14bd7103e8.png)
![Screenshot from 2020-08-20 10-55-44](https://user-images.githubusercontent.com/6207644/90788438-d05db380-e2d3-11ea-9844-e49cfe03cd91.png)
